homeassistant/components/elgato/light.py

```diff
@@ -89,17 +91,49 @@ def color_temp(self) -> int | None:
     @property
     def min_mireds(self) -> int:
         """Return the coldest color_temp that this light supports."""
+        # Elgato lights with color capabilities have a different lowest value
+        if self._settings.power_on_hue is not None:
+            return 153
         return 143
 
     @property
     def max_mireds(self) -> int:
         """Return the warmest color_temp that this light supports."""
+        # Elgato lights with color capabilities have a different highest value
+        if self._settings.power_on_hue is not None:
+            return 285
         return 344
 
     @property
-    def supported_features(self) -> int:
-        """Flag supported features."""
-        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+    def supported_color_modes(self) -> set[str]:
+        """Flag supported color modes."""
+        modes = [COLOR_MODE_COLOR_TEMP]
+
+        # Check if the light is capable of doing colors
+        if self._settings.power_on_hue is not None:
```

Comment:
Can we calculate all properties that rely on `self._settings` inside the constructor to avoid calculating them on every state change?

Reply:
I think it is ugly to have a big constructor doing all that stuff, maybe we should some kind of decorator that handles this?

Similar to `@lru_cache(...)` or `@singleton` kinda deal? `@once` or something?

Reply:
Pre-caled until we have the other option 👍